enricher/standard/src/main/test/java/io/fabric8/maven/enricher/standard/VolumePermissionEnricherTest.java

```diff
@@ -0,0 +1,126 @@
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.enricher.standard;
+
+import io.fabric8.kubernetes.api.model.ContainerBuilder;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.kubernetes.api.model.PodTemplate;
+import io.fabric8.kubernetes.api.model.PodTemplateBuilder;
+import io.fabric8.maven.core.config.ProcessorConfig;
+import io.fabric8.maven.enricher.api.BaseEnricher;
+import io.fabric8.maven.enricher.api.EnricherContext;
+import io.fabric8.utils.Strings;
+import mockit.Expectations;
+import mockit.Mocked;
+import mockit.integration.junit4.JMockit;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Collections;
+import java.util.TreeMap;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(JMockit.class)
+public class VolumePermissionEnricherTest {
+
+    @Mocked
+    private EnricherContext context;
+
+    // *******************************
+    // Tests
+    // *******************************
+
+    private static final class TestConfig {
+        private final String permission;
+        private final String initContainerName;
+        private final String[] volumeNames;
+
+        private TestConfig(String permission, String initContainerName, String... volumeNames) {
+            this.permission = permission;
+            this.initContainerName = initContainerName;
+            this.volumeNames = volumeNames;
+        }
+    }
+
+    @Test
+    public void testAdapt() throws Exception {
+        final TestConfig[] data = new TestConfig[]{
+            new TestConfig(null, null),
+            new TestConfig(null, "init", "volumeA"),
+            new TestConfig(null, "init", "volumeA", "volumeB")
+        };
+
+        for (final TestConfig tc : data) {
+            final ProcessorConfig config = new ProcessorConfig(null, null,
+                    Collections.singletonMap(VolumePermissionEnricher.ENRICHER_NAME, new TreeMap(Collections
+                            .singletonMap(VolumePermissionEnricher.Config.permission.name(), tc.permission))));
+
+            // Setup mock behaviour
+            new Expectations() {{ context.getConfig(); result = config; }};
+
+            VolumePermissionEnricher enricher = new VolumePermissionEnricher(context);
+
+            PodTemplateBuilder ptb = new PodTemplateBuilder().withNewMetadata().and()
```

Comment:
Just like `end...` but more fluent (reads better & means you don't have to remember what kind you're editing).